File: qc-core/src/main/java/com/eager/questioncloud/core/domain/question/QuestionJpaRepository.java
package com.eager.questioncloud.core.domain.question;

import java.util.Optional;
import org.springframework.data.jpa.repository.JpaRepository;

public interface QuestionJpaRepository extends JpaRepository<QuestionEntity, Long> {
    Optional<QuestionEntity> findByIdAndCreatorId(Long id, Long creatorId);
}

File: qc-infra/qc-rdb/src/main/kotlin/com/eager/questioncloud/question/repository/QuestionJpaRepository.kt
package com.eager.questioncloud.question.repository

import com.eager.questioncloud.question.entity.QuestionEntity
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface QuestionJpaRepository : JpaRepository<QuestionEntity, Long> {
    fun findByIdAndCreatorId(id: Long, creatorId: Long): Optional<QuestionEntity>
}
